replication.go

```diff
@@ -388,6 +388,9 @@ func (r *Raft) heartbeat(s *followerReplication, stopCh chan struct{}) {
 			case <-stopCh:
 			}
 		} else {
+			if failures > 0 {
+				r.observe(ResumedHeartbeatObservation{PeerID: s.peer.ID})
```

Comment:
Hi @dmitrime, thanks for this PR!  What do you think about adding `LastContact: s.LastContact()` like we do in the FailedHearbeatObservation?  Both for symmetry, and because it might be of interest to someone consuming that observation.

Reply:
Hi @ncabatoff, I've considered this in the first version I tried, but then noticed that `LastContact` was not always showing the correct timestamp. I believe that's due to the fact that it is being concurrently updated in some another place once the connection is re-established (`sendLatestSnapshot` or `replicateTo`).

In any case, `ResumedHeartbeatObservation` will likely be used in together with `FailedHeartbeatObservation` anyway, so it's not a problem for the client to save the timestamp of the first `FailedHeartbeatObservation` event and then use that later.

Reply:
Makes sense.  I agree it's not necessary.